clients/src/main/java/org/apache/kafka/common/requests/ShareFetchResponse.java

```diff
@@ -156,6 +163,23 @@ public static int recordsSize(ShareFetchResponseData.PartitionData partition) {
         return partition.records() == null ? 0 : partition.records().sizeInBytes();
     }
 
+    /**
+     * Creates a {@link org.apache.kafka.common.requests.ShareFetchResponse} from the given data.
+     * This method converts null records to {@link org.apache.kafka.common.record.MemoryRecords#EMPTY}
+     * to ensure consistent record representation in the response.
+     *
+     * <p><strong>This method should only be used in server-side.</strong></p>
+     */
+    public static ShareFetchResponse of(ShareFetchResponseData data) {
```

Comment:
Could we get rid of this constructor and direct all callers to the other `of` constructor?

Reply:
Yes.
I have just reviewed the current usage. Refactoring them to `ShareFetchResponse.of(Errors, int, LinkedHashMap, List)` doesn't take too much effort.